src/test/java/com/ibm/stocator/fs/swift2d/TestSwiftFileSystemBasicOps.java

```diff
@@ -0,0 +1,270 @@
+    } finally {
+      delete(fs, path);
+    }
+  }
+
+  /**
+   * Assert that if a directory that has children is deleted, it is still
+   * a directory
+   *
+   * @throws Throwable if not, or if something else failed
+   */
+  @Ignore("Not supported")
+  public void testDirStaysADir() throws Throwable {
+    Path path = new Path("/test/dirStaysADir");
+    Path child = new Path(path, "child");
+    try {
+      //create the dir
+      fs.mkdirs(path);
+      //assert the parent has the directory nature
+      SwiftTestUtils.assertIsDirectory(fs, path);
+      //create the child dir
+      writeTextFile(fs, child, "child file", true);
+      //assert the parent has the directory nature
+      SwiftTestUtils.assertIsDirectory(fs, path);
+      //now rm the child
+      delete(fs, child);
+    } finally {
+      deleteR(fs, path);
+    }
+  }
+
+  @Ignore("Not supported")
+  public void testCreateMultilevelDir() throws Throwable {
+    Path base = new Path(getBaseURI() + "/test/CreateMultilevelDir");
+    Path path = new Path(base, "1/2/3");
+    fs.mkdirs(path);
+    assertExists("deep multilevel dir not created", path);
+    fs.delete(base, true);
+    assertPathDoesNotExist("Multilevel delete failed", path);
+    assertPathDoesNotExist("Multilevel delete failed", base);
+
+  }
+
+  @Ignore("Not supported")
+  public void testCreateDirWithFileParent() throws Throwable {
+    Path path = new Path(getBaseURI() + "/test/CreateDirWithFileParent");
+    Path child = new Path(path, "subdir/child");
+    fs.mkdirs(path.getParent());
+    try {
+      //create the child dir
+      writeTextFile(fs, path, "parent", true);
+      try {
+        fs.mkdirs(child);
+      } catch (ParentNotDirectoryException expected) {
+        LOG.debug("Expected Exception", expected);
+      }
+    } finally {
+      fs.delete(path, true);
+    }
+  }
+
+  @Ignore("Unexpected")
+  public void testLongObjectNamesForbidden() throws Throwable {
+    StringBuilder buffer = new StringBuilder(1200);
+    buffer.append("/");
+    for (int i = 0; i < (1200 / 4); i++) {
+      buffer.append(String.format("%04x", i));
+    }
+    String pathString = buffer.toString();
+    pathString = "/A/B/C";
+    Path path = new Path(getBaseURI() + pathString);
+    try {
+      writeTextFile(fs, path, pathString, true);
+      //if we get here, problems.
+      fs.delete(path, false);
+      fail("Managed to create an object with a name of length "
+              + pathString.length());
+    } catch (Exception e) {
+      //expected
+      //LOG.debug("Caught exception " + e, e);
+    }
+  }
+
+  @Ignore("Unexpected")
+  public void testLsNonExistentFile() throws Exception {
+    try {
+      Path path = new Path(getBaseURI() + "/test/hadoop/file");
+      FileStatus[] statuses = fs.listStatus(path);
+      fail("Should throw FileNotFoundException on " + path
+              + " but got list of length " + statuses.length);
+    } catch (FileNotFoundException fnfe) {
+      // expected
```

Comment:
For these types of test where we use exceptions we can put the exception to be expected in the annotation.
http://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests